tools/ci/check_grep.sh

```diff
@@ -12,288 +12,357 @@ NC="\033[0m" # No Color
     st=1
 fi;
-if grep -Pzo '"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/reinforced/plasma/plastitanium,\n[^)]*?/area/.+\)' _maps/**/*.dmm;	then
+if rg -U '"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/reinforced/plasma/plastitanium,\n[^)]*?/area/.+\)' -g *.dmm;	then
 	echo
     echo -e "${RED}ERROR: Found grille above a fulltile plastitanium window. Please replace it with the proper structure spawner.${NC}"
     st=1
 fi;
-if grep -Pzo '/obj/machinery/conveyor/inverted[/\w]*?\{\n[^}]*?dir = [1248];[^}]*?\},?\n' _maps/**/*.dmm;	then
+part "converyor cardinals inversion"
+if rg -U '/obj/machinery/conveyor/inverted[/\w]*?\{\n[^}]*?dir = [1248];[^}]*?\},?\n' -g *.dmm;	then
 	echo
     echo -e "${RED}ERROR: Found an inverted conveyor belt with a cardinal dir. Please replace it with a normal conveyor belt.${NC}"
     st=1
 fi;
-if grep -P '^/area/.+[\{]' _maps/**/*.dmm;	then
+part "area varedits"
+if rg '^/area/.+[\{]' -g *.dmm;	then
 	echo
     echo -e "${RED}ERROR: Variable editted /area path use detected in a map, please replace with a proper area path.${NC}"
     st=1
 fi;
-if grep -P '\/turf\s*[,\){]' _maps/**/*.dmm; then
+part "base turf type"
+if rg '/turf\s*[,\){]' -g *.dmm; then
 	echo
     echo -e "${RED}ERROR: Base /turf path use detected in maps, please replace it with a proper turf path.${NC}"
     st=1
 fi;
-if grep -Pzo '"\w+" = \(\n[^)]*?/turf/[/\w]*?,\n[^)]*?/turf/[/\w]*?,\n[^)]*?/area/.+?\)' _maps/**/*.dmm; then
+part "multiple turfs"
+if rg -U '"\w+" = \(\n[^)]*?/turf/[/\w]*?,\n[^)]*?/turf/[/\w]*?,\n[^)]*?/area/.+?\)' -g *.dmm; then
 	echo
     echo -e "${RED}ERROR: Multiple turfs detected on the same tile! Please choose only one turf!${NC}"
     st=1
 fi;
-if grep -Pzo '"\w+" = \(\n[^)]*?/area/.+?,\n[^)]*?/area/.+?\)' _maps/**/*.dmm; then
+part "multiple areas"
+if rg -U '"\w+" = \(\n[^)]*?/area/.+?,\n[^)]*?/area/.+?\)' -g *.dmm; then
 	echo
     echo -e "${RED}ERROR: Multiple areas detected on the same tile! Please choose only one area!${NC}"
     st=1
 fi;
-if grep -ni 'nanotransen' _maps/**/*.dmm; then
+part "common spelling mistakes"
+if rg -ni 'nanotransen' -g *.dmm; then
 	echo
     echo -e "${RED}ERROR: Misspelling of Nanotrasen detected in maps, please remove the extra N(s).${NC}"
     st=1
 fi;
-if grep -i 'centcomm' _maps/**/*.dmm; then
+if rg -i'centcomm' -g *.dmm; then
 	echo
     echo -e "${RED}ERROR: Misspelling(s) of CentCom detected in maps, please remove the extra M(s).${NC}"
     st=1
 fi;
+part "docking_port varedits"
+if rg -U '^/obj/docking_port/mobile.*\{\n[^}]*(width|height|dwidth|dheight)[^}]*\}' -g *.dmm; then
+	echo
+	echo -e "${RED}ERROR: Custom mobile docking_port sizes detected. This is done automatically and should not be varedits.${NC}"
+	echo -e "\t\t${RED}Please remove the width, height, dwidth, and dheight varedits from the docking_port.${NC}"
```

Comment:
```suggestion
	echo -e "${RED}ERROR: Custom mobile docking_port sizes detected. This is done automatically and should not be varedits."
	echo -e "\t\tPlease remove the width, height, dwidth, and dheight varedits from the docking_port.${NC}"
```

No need to undefine and redefine the color since both lines are going to be red. Also, `\t` works here? Real if big